packages/synapse-react-client/src/utils/hooks/useUploadFileEntity/usePrepareFileEntityUpload.ts

```diff
@@ -4,17 +4,24 @@ import { useSynapseContext } from '../../context/index'
 import { getFileEntityIdWithSameName } from './getFileEntityIdWithSameName'
 import { useCreatePathsAndGetParentId } from './useCreatePathsAndGetParentId'
 
-export type FilePreparedForUpload = {
-  file: File
-  parentId: string
-  existingEntityId: string | null
-}
+export type NewEntityFileUploadArgs = { file: File; rootContainerId: string }
+export type NewEntityFileUploadReturn = { file: File; parentId: string }
+export type UpdateEntityFileUpload = { file: File; existingEntityId: string }
+
+export type FilePreparedForUpload =
+  | NewEntityFileUploadReturn
+  | UpdateEntityFileUpload
 
 export type PrepareDirsForUploadReturn = {
-  newFileEntities: FilePreparedForUpload[]
-  updatedFileEntities: FilePreparedForUpload[]
+  filesReadyForUpload: FilePreparedForUpload[]
+  filesToPromptForNewVersion: FilePreparedForUpload[]
```

Comment:
Renamed these properties to more accurately capture the purpose of this hook. With the changes in this PR, the old names are now inaccurate. If a file is explicitly uploaded as a new version of an entity, then there's no need to show a prompt.